addons/Connection/src/presenter.js

```diff
@@ -335,17 +411,121 @@ function AddonConnection_create() {
     }
 
+    presenter.addDisabledElementsFromInitialValues = function () {
+        presenter.initialValues.forEach(function (initialValue) {
+           if (initialValue.isDisabled) {
+               presenter.disabledConnections.push({
+                   id1: initialValue.from,
+                   id2: initialValue.to
+               });
+           }
+        });
+    };
+
+
+
+    presenter.getInitialValues = function (model) {
+        var modelValidator = new ModelValidator();
+        var validated = modelValidator.validate(model, [
+            ModelValidators.List("initialConnections", [
+                ModelValidators.String("from", {default: "", trim: false}),
+                ModelValidators.String("to", {default: "", trim: false}),
+                ModelValidators.Boolean("isDisabled", {default: false})
+            ]),
+            ModelValidators.String("disabledConnectionColor", {default: "", trim: true})
+        ]);
+
+        return {
+            initialConnections: validated.value.initialConnections,
+            disabledConnectionColor: validated.value.disabledConnectionColor
+        };
+    };
+
+    /**
+     * @param initialValue {{from: string, to: string}}
+     */
+    presenter.drawInitialValue = function (initialValue) {
+        function containsID (array, idToFind, toReturn) {
+            for (var i = 0; i < array.length; i++) {
+                if (array[i].id === idToFind) {
+                    return toReturn;
+                }
+            }
+
+            return false;
+        }
+
+        function oneOfValueIsEmpty () {
+            return initialValue.from.trim() === "" && initialValue.to.trim() === "";
+        }
+
+        function areFromDifferentCols() {
+            var fromColumn = containsID(presenter.model['Left column'], initialValue.from, presenter.ELEMENT_SIDE.LEFT) ||
+                containsID(presenter.model['Right column'], initialValue.from, presenter.ELEMENT_SIDE.RIGHT);
+
+            var toColumn = containsID(presenter.model['Left column'], initialValue.to, presenter.ELEMENT_SIDE.LEFT) ||
+                containsID(presenter.model['Right column'], initialValue.to, presenter.ELEMENT_SIDE.RIGHT);
+
+            return fromColumn !== toColumn;
+        }
+
+        function bothExists () {
+            var fromExists = containsID(presenter.model['Left column'], initialValue.from, true) ||
+                containsID(presenter.model['Right column'], initialValue.from, true);
+
+            var toExists = containsID(presenter.model['Left column'], initialValue.to, true) ||
+                containsID(presenter.model['Right column'], initialValue.to, true);
+
+            return fromExists && toExists;
+        }
+
+        if (!oneOfValueIsEmpty() && bothExists() && areFromDifferentCols()) {
+            pushConnection(new Line(getElementById(initialValue.from), getElementById(initialValue.to)), false);
+            return true;
+        }
+
+        if (!oneOfValueIsEmpty()) {
```

Comment:
Czy w przypadku gdy jedna z wartości jest pusta (oneOfValueIsEmpty() == true) nie powinien zostać wyświetlony komunikat o błędzie?
W teorii, to powinno zostać wyłapane już w momencie walidacji modelu, więc jest to jakaś sytuacja wyjatkowa.

Reply:
Jeżeli jedna z wartości jest pusta to nie biorę pod uwagę takiego wpisu.